### PR TITLE
Feature/hdmi support

### PR DESCRIPTION
**Pull request by Nathoufresh**

### CODE REVIEW

File: src/device.hpp [hunk 1]
```diff
@@ -46,10 +56,10 @@ namespace Deltacast
         void enable_loopback(int index);
         void disable_loopback(int index);
         bool wait_for_incoming_signal(int rx_index, const std::atomic_bool& stop_is_requested);
-        SignalInformation get_incoming_signal_information(int rx_index);
 
-        bool wait_genlock_locked(const std::atomic_bool& stop_is_requested);
-        bool configure_genlock(int genlock_source_rx_index, SignalInformation signal_info);
+        bool wait_sync_locked(const std::atomic_bool& stop_is_requested, std::unique_ptr<Helper::VideoInformation>& video_info);
+        bool configure_sync(int genlock_source_rx_index, std::unique_ptr<Helper::VideoInformation>& video_info);
+        std::unique_ptr<Helper::VideoInformation> get_video_information_for_channel(int index, Direction direction);
```
Comment (laurent-radoux) [inline src/device.hpp:62]: As it stands for a factory method, I would suggest a different name to reflect that it creates an instance of a video_information based on the underlying technology of the channel.

File: cmake/FindVideoMaster.cmake
Comment (laurent-radoux) [inline cmake/FindVideoMaster.cmake:1]: Let's move to find_package with newer version of VideoMaster

File: src/device.cpp [hunk 2]
```diff
@@ -22,300 +23,336 @@
...
-    {8, VHD_CORE_BP_RX8_STATUS},
-    {9, VHD_CORE_BP_RX9_STATUS},
-    {10, VHD_CORE_BP_RX10_STATUS},
-    {11, VHD_CORE_BP_RX11_STATUS},
+   { 0, VHD_CORE_BP_RX0_STATUS }, { 1, VHD_CORE_BP_RX1_STATUS },   { 2, VHD_CORE_BP_RX2_STATUS },
+   { 3, VHD_CORE_BP_RX3_STATUS }, { 4, VHD_CORE_BP_RX4_STATUS },   { 5, VHD_CORE_BP_RX5_STATUS },
+   { 6, VHD_CORE_BP_RX6_STATUS }, { 7, VHD_CORE_BP_RX7_STATUS },   { 8, VHD_CORE_BP_RX8_STATUS },
+   { 9, VHD_CORE_BP_RX9_STATUS }, { 10, VHD_CORE_BP_RX10_STATUS }, { 11, VHD_CORE_BP_RX11_STATUS },
 };
 const std::unordered_map<uint32_t, VHD_SDI_BOARDPROPERTY> id_to_rx_video_standard_prop = {
```
Comment (laurent-radoux) [inline src/device.cpp:64]: No longer useful thanks to ApiHelper.

File: src/device.cpp [hunk 3]
```diff
@@ -22,300 +23,336 @@
...
-    {10, VHD_SDI_BP_RX10_STANDARD},
-    {11, VHD_SDI_BP_RX11_STANDARD},
+   { 0, VHD_SDI_BP_RX0_STANDARD },   { 1, VHD_SDI_BP_RX1_STANDARD },
+   { 2, VHD_SDI_BP_RX2_STANDARD },   { 3, VHD_SDI_BP_RX3_STANDARD },
+   { 4, VHD_SDI_BP_RX4_STANDARD },   { 5, VHD_SDI_BP_RX5_STANDARD },
+   { 6, VHD_SDI_BP_RX6_STANDARD },   { 7, VHD_SDI_BP_RX7_STANDARD },
+   { 8, VHD_SDI_BP_RX8_STANDARD },   { 9, VHD_SDI_BP_RX9_STANDARD },
+   { 10, VHD_SDI_BP_RX10_STANDARD }, { 11, VHD_SDI_BP_RX11_STANDARD },
 };
 const std::unordered_map<uint32_t, VHD_SDI_BOARDPROPERTY> id_to_rx_clock_divisor_prop = {
```
Comment (laurent-radoux) [inline src/device.cpp:72]: No longer useful thanks to ApiHelper.

File: src/device.cpp [hunk 4]
```diff
@@ -22,300 +23,336 @@
...
-    {10, VHD_SDI_BP_RX10_CLOCK_DIV},
-    {11, VHD_SDI_BP_RX11_CLOCK_DIV},
+   { 0, VHD_SDI_BP_RX0_CLOCK_DIV },   { 1, VHD_SDI_BP_RX1_CLOCK_DIV },
+   { 2, VHD_SDI_BP_RX2_CLOCK_DIV },   { 3, VHD_SDI_BP_RX3_CLOCK_DIV },
+   { 4, VHD_SDI_BP_RX4_CLOCK_DIV },   { 5, VHD_SDI_BP_RX5_CLOCK_DIV },
+   { 6, VHD_SDI_BP_RX6_CLOCK_DIV },   { 7, VHD_SDI_BP_RX7_CLOCK_DIV },
+   { 8, VHD_SDI_BP_RX8_CLOCK_DIV },   { 9, VHD_SDI_BP_RX9_CLOCK_DIV },
+   { 10, VHD_SDI_BP_RX10_CLOCK_DIV }, { 11, VHD_SDI_BP_RX11_CLOCK_DIV },
 };
 const std::unordered_map<uint32_t, VHD_SDI_BOARDPROPERTY> id_to_rx_interface_prop = {
```
Comment (laurent-radoux) [inline src/device.cpp:80]: No longer useful thanks to ApiHelper.

File: src/device.cpp [hunk 5]
```diff
@@ -22,300 +23,336 @@
...
-    {10, VHD_SDI_BP_RX10_INTERFACE},
-    {11, VHD_SDI_BP_RX11_INTERFACE},
+   { 0, VHD_SDI_BP_RX0_INTERFACE },   { 1, VHD_SDI_BP_RX1_INTERFACE },
+   { 2, VHD_SDI_BP_RX2_INTERFACE },   { 3, VHD_SDI_BP_RX3_INTERFACE },
+   { 4, VHD_SDI_BP_RX4_INTERFACE },   { 5, VHD_SDI_BP_RX5_INTERFACE },
+   { 6, VHD_SDI_BP_RX6_INTERFACE },   { 7, VHD_SDI_BP_RX7_INTERFACE },
+   { 8, VHD_SDI_BP_RX8_INTERFACE },   { 9, VHD_SDI_BP_RX9_INTERFACE },
+   { 10, VHD_SDI_BP_RX10_INTERFACE }, { 11, VHD_SDI_BP_RX11_INTERFACE },
 };
 const std::unordered_map<uint32_t, VHD_GENLOCKSOURCE> id_to_rx_genlock_source = {
```
Comment (laurent-radoux) [inline src/device.cpp:88]: No longer useful thanks to ApiHelper.

File: src/device.cpp [hunk 6]
```diff
@@ -22,300 +23,336 @@
...
 
-        ULONG status = VHD_CORE_RXSTS_UNLOCKED;
-        auto api_success = ApiSuccess{VHD_GetBoardProperty(*handle(), id_to_rx_status_prop.at(rx_index), &status)};
-        if (api_success && !(status & VHD_CORE_RXSTS_UNLOCKED))
-            return true;
-    }
+      ULONG status = VHD_SDI_GNLKSTS_NOREF | VHD_SDI_GNLKSTS_UNLOCKED;
+      auto  api_success = ApiSuccess{ VHD_GetBoardProperty(*handle(), genlock_status_prop_optional.value(), &status) };
+      if (api_success && !(status & VHD_SDI_GNLKSTS_NOREF) && !(status & VHD_SDI_GNLKSTS_UNLOCKED))
+         return true;
```
Comment (laurent-radoux) [inline src/device.cpp:221]: Move the sleep at the end of the loop.

File: src/device.cpp [hunk 7]
```diff
@@ -22,300 +23,336 @@
...
-        ULONG driver_version = 0, firmware_version, number_of_rx_channels, number_of_tx_channels;
+std::ostream& operator<<(std::ostream& os, const Device& device)
+{
+   ULONG driver_version = 0, firmware_version, number_of_rx_channels, number_of_tx_channels;
 
-        VHD_GetBoardProperty(**(device._device_handle), VHD_CORE_BP_DRIVER_VERSION, &driver_version);
-        VHD_GetBoardProperty(**(device._device_handle), VHD_CORE_BP_FIRMWARE_VERSION, &firmware_version);
-        VHD_GetBoardProperty(**(device._device_handle), VHD_CORE_BP_NB_RXCHANNELS, &number_of_rx_channels);
-        VHD_GetBoardProperty(**(device._device_handle), VHD_CORE_BP_NB_TXCHANNELS, &number_of_tx_channels);
+   VHD_GetBoardProperty(**(device._device_handle), VHD_CORE_BP_DRIVER_VERSION, &driver_version);
```
Comment (laurent-radoux) [inline src/device.cpp:327]: We can use the new VHD_GetDriverStringVersion

**Review by laurent-radoux — CHANGES_REQUESTED** [review]

File: src/device.hpp (hunk 1 above)
Comment (Nathoufresh) [reply src/device.hpp:62]: Something like : factory_create_video_information_for_channel ?

**Pull request closed without merging**